lib/upload.ts

```diff
@@ -215,6 +230,9 @@ export class BaseUpload {
       }
     }
 
+    // Setup stall detection before starting the upload
+    this._setupStallDetection()
```

Comment:
This PR currently uses one stall detector for the entire upload, which seems to increase the complexity of the stall detection as it now has to support pausing and resuming as well.

A simpler alternative would be to use to stall detector per HTTP request that carries upload data (i.e. PATCH or POST requests with `uploadDataDuringCreation`). Then the stall detector has to keep less state as we just discard it when the request finishes or is aborted, but we should still be capable of detecting stalled uploads.

Reply:
@Acconut 

This should now have been addressed along with the other feedback. Feel free to take another look.

❤️